### DISCUSSION TITLE
Does Highs_setSolution support partial/"integer only" solutions?

### DISCUSSION THREAD

**Discussion by jannicklange** [ERGO-Code/HiGHS#979] [Q&A] [answered]
It seems to me, as though [Highs_setSolution](https://github.com/ERGO-Code/HiGHS/blob/c0e1182bd350d4ecfc4600aee3e7ff71aacae6cd/src/interfaces/highs_c_api.h#L870) expects a complete solution to be passed. I do have a MIP model, that contains ~4.6k binary variables. When I pass a feasible assignment of those variables via `Highs_setSolution`, HiGHS prints the following log (and starts to solve without an initial incumbent):

```
Running HiGHS 1.2.2 [date: 2022-10-13, git hash: be72abd29]
Copyright (c) 2022 ERGO-Code under MIT licence terms
Presolving model
57947 rows, 45710 cols, 417766 nonzeros
28738 rows, 25106 cols, 147857 nonzeros
14381 rows, 20850 cols, 68936 nonzeros
11693 rows, 18162 cols, 64418 nonzeros
8163 rows, 14421 cols, 56048 nonzeros
7859 rows, 14190 cols, 55262 nonzeros

MIP start solution is infeasible, objective value is 3252.36

Solving MIP model with:
   7859 rows
   14190 cols (965 binary, 0 integer, 0 implied int., 13225 continuous)
   55262 nonzeros

        Nodes      |    B&B Tree     |            Objective Bounds              |  Dynamic Constraints |       Work
     Proc. InQueue |  Leaves   Expl. | BestBound       BestSol              Gap |   Cuts   InLp Confl. | LpIters     Time

         0       0         0   0.00%   724.6079188     inf                  inf        0      0      0         0    17.7s

```

When I pass the same model/start values to Gurobi, the solution is accepted (+ the LP/continuous part is automatically derived):

```
Optimize a model with 404848 rows, 294984 columns and 1212971 nonzeros
Model fingerprint: 
Variable types: 290370 continuous, 4614 integer (4614 binary)
Coefficient statistics:
  Matrix range     [1e-01, 2e+05]
  Objective range  [2e-02, 6e+02]
  Bounds range     [1e+00, 1e+00]
  RHS range        [1e+00, 2e+04]

User MIP start produced solution with objective 38862.6 (2.18s)
Loaded user MIP start with objective 38862.6
Processed MIP start in 2.22 seconds (0.52 work units)

Presolve removed 398603 rows and 283302 columns (presolve time = 5s) ...
Presolve removed 398607 rows and 283306 columns
Presolve time: 5.31s
Presolved: 6241 rows, 11678 columns, 43423 nonzeros
Variable types: 10877 continuous, 801 integer (801 binary)

Root simplex log...

Iteration    Objective       Primal Inf.    Dual Inf.      Time
       0    6.6422287e+03   7.168917e+03   0.000000e+00      8s
    2602    3.0150943e+04   0.000000e+00   0.000000e+00      8s

Root relaxation: objective 3.015094e+04, 2602 iterations, 0.11 seconds (0.03 work units)

    Nodes    |    Current Node    |     Objective Bounds      |     Work
 Expl Unexpl |  Obj  Depth IntInf | Incumbent    BestBd   Gap | It/Node Time

     0     0 30150.9431    0   83 38862.5947 30150.9431  22.4%     -    8s

```

I pass the solution to [Highs_setSolution](https://github.com/ERGO-Code/HiGHS/blob/c0e1182bd350d4ecfc4600aee3e7ff71aacae6cd/src/interfaces/highs_c_api.h#L870) in a `double[]` with size `#variablesInModel` as `col_value` parameter, where all non-binary/integer fields are set to 0. The other 3 params are set to `null`.

Is there an alternative way to set the solution, e.g. with an "index-filter"?
Might the problem be caused because HiGHS only tries to load the start solution after the presolve, where most of the (binary) variables were already removed from the model?
Is there any other way to pass an initial (partial) solution to HiGHS?

Thanks,
Jannick

**Comment by jajhall** [comment, marked as answer]
I've had another query about specifying an initial solution (via the command line) so will give this some thought soon

**Reply by jajhall** [reply]
This facility will be available when #1013 is merged into master

**Reply by jannicklange** [reply]
Thanks for the Update. I just checked out the latest release, and my example solution was imported successfully :)
